mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -556,6 +571,7 @@ class Word2VecModel private[spark] (
       .sortBy(- _._2)
       .take(num + 1)
       .tail
+      .map(v => (if (vecNorm == 0) v else (v._1, v._2 / vecNorm)))
```

Comment:
I agree that we should define cosine similarity with a zero vector to be 0. In this case the results are pretty meaningless anyway, since the dot product was already 0 for everything, and so the top N are random. I'd say:

```
.map { case (word, cosVec) => (word, if (vecNorm == 0.0) 0.0 else (cosVec / vecNorm)) }
```
For more efficiency we could only apply this map in the corner case that `vecNorm` is 0.0; not sure if it's simpler but it's a little less work:

```
val result = wordList....tail
if (vecNorm != 0.0) [
  result = result.map { case (word, cosVec) => (word, cosVec / vecNorm) }
}
result.toArray
```